session/sql_store.go

```diff
@@ -0,0 +1,839 @@
+		remoteKey   = remotePubKey.SerializeCompressed()
+	)
+	return s.db.ExecTx(ctx, &writeTxOpts, func(db SQLQueries) error {
+		id, err := db.GetSessionIDByAlias(ctx, alias[:])
+		if err != nil {
+			return fmt.Errorf("unable to get session Alias: %v",
+				err)
+		}
+
+		return db.SetSessionRemotePublicKey(
+			ctx, sqlc.SetSessionRemotePublicKeyParams{
+				ID:              id,
+				RemotePublicKey: remoteKey,
+			},
+		)
+	})
+}
+
+// getSqlUnusedAliasAndKeyPair can be used to generate a new, unused, local
+// private key and session Alias pair. Care must be taken to ensure that no
+// other thread calls this before the returned Alias and key pair from this
+// method are either used or discarded.
+//
+// NOTE: This is part of the Store interface.
+func getSqlUnusedAliasAndKeyPair(ctx context.Context, db SQLQueries) (ID,
+	*btcec.PrivateKey, error) {
+
+	// Spin until we find a key with an Alias that does not collide
+	// with any of our existing IDs.
+	for {
+		// Generate a new private key and Alias pair.
+		privKey, alias, err := NewSessionPrivKeyAndID()
+		if err != nil {
+			return ID{}, nil, err
+		}
+
+		// Check that no such legacy Alias exits.
+		_, err = db.GetSessionByAlias(ctx, alias[:])
+		if errors.Is(err, sql.ErrNoRows) {
+			return alias, privKey, nil
+		} else if err != nil {
+			return ID{}, nil, fmt.Errorf("unable to get "+
+				"session: %w", err)
+		}
+
+		continue
+	}
+}
+
+// GetSession returns the session with the given legacy Alias.
+//
+// NOTE: This is part of the Store interface.
+func (s *SQLStore) GetSession(ctx context.Context, alias ID) (*Session, error) {
+	var (
+		readTxOpts = db.NewQueryReadTx()
+		sess       *Session
+	)
+	err := s.db.ExecTx(ctx, &readTxOpts, func(db SQLQueries) error {
+		dbSess, err := db.GetSessionByAlias(ctx, alias[:])
+		if errors.Is(err, sql.ErrNoRows) {
+			return ErrSessionNotFound
+		} else if err != nil {
+			return fmt.Errorf("unable to get session: %w", err)
+		}
+
+		sess, err = unmarshalSession(ctx, s.db, dbSess)
+		if err != nil {
+			return fmt.Errorf("unable to unmarshal session: %w",
+				err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return sess, nil
+}
+
+// GetGroupID will return the legacy group Alias for the given legacy session Alias.
+//
+// NOTE: This is part of the AliasToGroupIndex interface.
+func (s *SQLStore) GetGroupID(ctx context.Context, sessionID ID) (ID, error) {
+	var (
+		readTxOpts    = db.NewQueryReadTx()
+		legacyGroupID ID
+	)
+	err := s.db.ExecTx(ctx, &readTxOpts, func(db SQLQueries) error {
+		// Get the session using the legacy Alias.
+		sess, err := db.GetSessionByAlias(ctx, sessionID[:])
+		if err != nil {
+			return ErrUnknownGroup
+		}
+
+		if !sess.GroupID.Valid {
+			return fmt.Errorf("session does not have a group Alias")
+		}
+
+		// Get the legacy group Alias using the session group Alias.
+		legacyGroupIDB, err := db.GetAliasBySessionID(
+			ctx, sess.GroupID.Int64,
+		)
+		if err != nil {
+			return err
+		}
```

Comment:
Maybe return `ErrSessionNotFound` on no rows, but I think that'd be unreachable.